packages/ui-components/src/components/toggle-tip/toggle-tip.tsx

```diff
@@ -0,0 +1,150 @@
+import { isEmpty } from 'lodash-es';
+import { ETooltipPosition } from '../../utils/types/components';
+import { DEFAULT_AUTOPLACEMENT_CONFIG, DEFAULT_OFFSET, DEFAULT_SHIFT_CONFIG, DEFAULT_TOOLTIP_CONFIG } from '../tooltip/tooltip.config';
+import { getArrowElementPositionConfig, OFFSET_WITH_ARROW } from './config';
+import { IToggleTip, IToggleTipEvents } from './toggle-tip.types';
+
+/**
+ * @part toggle-tip-container - The toggle tip container.
+ * @part toggle-tip-slot-content - The toggle tip slot content.
+ */
+@Component({
+	tag: 'kv-toggle-tip',
+	styleUrl: 'toggle-tip.scss',
+	shadow: true
+})
+export class KvToggleTip implements IToggleTip, IToggleTipEvents {
+	/** @inheritdoc */
+	@Prop({ reflect: true }) text?: string;
+	/** @inheritdoc */
+	@Prop({ reflect: true }) position?: ETooltipPosition;
+	/** @inheritdoc */
+	@Prop({ reflect: true }) allowedPositions?: ETooltipPosition[];
+	/** @inheritdoc */
+	@Prop({ reflect: true }) options?: Partial<ComputePositionConfig>;
+	/** @inheritdoc */
+	@Prop({ reflect: true, mutable: true }) isOpen?: boolean = false;
+	/** @inheritdoc */
+	@Prop({ reflect: true }) isFixed?: boolean = false;
+	/** @inheritdoc */
+	@Prop({ reflect: true }) withArrow?: boolean = false;
+	/** @inheritdoc */
+	@Prop({ reflect: true }) disabled?: boolean = false;
+
+	/** @inheritdoc */
+	@Event() openStateChange: EventEmitter<boolean>;
+
+	/** The Host's element reference */
+	@Element() el: HTMLKvToggleTipElement;
+
+	@Listen('click', { target: 'window' })
+	checkForClickOutside(ev: { target: Node }) {
+		if (!this.isFixed) {
+			if (this.el.contains(ev.target)) {
+				return;
+			}
+			if (this.isOpen) {
+				this.isOpen = false;
+				this.addButtonClickEvent();
+				this.openStateChange.emit(false);
+			}
+		}
+	}
+
+	componentDidRender() {
+		this.calculateTogglePositions();
+	}
+
+	private getToggleOpenElement = (): HTMLElement | null => {
+		return this.el.shadowRoot.querySelector('#toggle-tip-open-element-wrapper') as HTMLElement | null;
+	};
+
+	private getToggleTipContainerElement = (): HTMLElement | null => {
+		return this.el.shadowRoot.querySelector('#toggle-tip-container') as HTMLElement | null;
+	};
+
+	private getToggleTipArrowElement = (): HTMLElement | null => {
+		return this.el.shadowRoot.querySelector('#toggle-tip-arrow') as HTMLElement | null;
+	};
+
+	private getMiddlewareConfig = () => {
+		const arrowElement = this.getToggleTipArrowElement();
+
+		const offSet = this.withArrow ? OFFSET_WITH_ARROW : DEFAULT_OFFSET;
+		const middleware = [offset(offSet), shift(DEFAULT_SHIFT_CONFIG)];
+		if (this.withArrow) middleware.push(arrow({ element: arrowElement, padding: 5 }));
+
+		return middleware;
+	};
+
+	private addButtonClickEvent = () => {
+		const toggleIcon = this.getToggleOpenElement();
+
+		const onButtonClick = () => {
+			this.isOpen = !this.isOpen;
+			this.openStateChange.emit(this.isOpen);
+		};
+		toggleIcon.addEventListener('mousedown', onButtonClick, { once: true });
```

Comment:
Without once true, toggle tip needs double click to open

Reply:
This may do for now as a "hacky" workaround, however I suspect this could have something to do with event propagation.

As per the [docs](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener#once) `once` removes the event listener after the first event fires so I'm not sure if this solution is sustainable in the long run.

Reply:
@pedrorebelocosta thanks 🙇